core/ppu_bg.go

```diff
@@ -176,11 +188,107 @@ func (ppu *PPU) tileFromBackground(background uint8, x uint16, y uint16) tile {
 	// See: https://wiki.superfamicom.org/backgrounds
 	raw := bit.JoinUint16(ppu.vram.bytes[addr], ppu.vram.bytes[addr+1])
 
-	return tile{
+	hSize, vSize := bg.tileSize()
+	colorDepth := ppu.colorDepth(background)
+	tileNumber := raw & 0x3FF
+
+	return bgTile{
+		baseTile: baseTile{
+			palette:    uint8((raw >> 10) & 0x7),
+			addr:       uint16(bg.tileSetBaseAddr)<<13 + uint16(tileNumber)*baseTileSize(colorDepth),
+			colorDepth: ppu.colorDepth(background),
+		},
 		vFlip:    raw&0x8000 != 0,
 		hFlip:    raw&0x4000 != 0,
 		priority: raw&0x2000 != 0,
-		palette:  uint8((raw >> 10) & 0x7),
-		number:   raw & 0x3ff,
+		hSize:    hSize,
+		vSize:    vSize,
+	}
+}
+
+//tileSize returns the size in pixel of tiles in the background
+func (bg *bg) tileSize() (uint16, uint16) {
+	hSize, vSize := uint16(8), uint16(8)
+	if bg.tileSizeFlag {
+		hSize, vSize = 16, 16
+	}
+
+	return hSize, vSize
+}
+
+// colorDepth returns the number of bits used for the colors in the background
```

Comment:
nit: can we add this to the doc:
```Mode    # Colors for BG
         1   2   3   4
======---=---=---=---=
0        4   4   4   4
1       16  16   4   -
2       16  16   -   -
3      256  16   -   -
4      256   4   -   -
5       16   4   -   -
6       16   -   -   -
7      256   -   -   -
7EXTBG 256 128   -   -```

From: https://wiki.superfamicom.org/backgrounds

Reply:
:+1: